### PR TITLE
Edited the README.md to clarify some of the "Gottchas"

### PR DESCRIPTION
**Pull request by bmeike**
... and to separate connected tests that require a server and those that do not

Also raise the timeout for the connectedUnitTests from 7 to 10 seconds (necessary for Nexus 4, 5.1.1)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,38 +70,41 @@ In case you don't want to use the precompiled version, you can build Realm yours
...
 
- * If you want to build with Android Studio, `ndk.dir` has to be defined in the `realm/local.properties` as well.
+ * If you want to build with Android Studio, `ndk.dir` has to be defined in the `realm/local.properties` as well.  Note that there is a `local.properites` in the root directory that is *not* the one that needs to be edited.  Again, presuming you used brew to install the NDK:
 
     ```
     ndk.dir=/usr/local/Cellar/android-ndk-r10e/r10e
     ```
 
- * If you are using OS X, you'd be better to add following lines to `~/.profile` (or `~/.zprofile` if the login shell is `zsh`) in order for Android Studio to see those environment variables.
+ * If you are using OS X, you must also define, export and launchctl the following two symbols, so that Android Studio can see them.
```
Comment (cmelchior) [inline README.md]: This is only true if you launch Android Studio from the launchpad, if you do it using the terminal it isn't needed. Also I believe the are issues with other shells like zsh, which @nhachicha uses.

That said, most people probably launch from the launchpad, this wording is probably fine.

File: README.md [hunk 2]
```diff
@@ -131,16 +134,18 @@ That command will generate:
  * a jar file for the annotations
  * a jar file for the annotations processor
 
+The full build may take an hour or more, to complete.
```
Comment (cmelchior) [inline README.md]: Good idea to have this

File: README.md (hunk 1 above)
Comment (bmeike) [inline README.md]: Good point.  Fixed.

File: README.md [hunk 3]
```diff
@@ -151,18 +156,37 @@ The repository is organized in six Gradle projects:
...
 
 The `./examples` folder contain a number of example projects showing how Realm can be used. If this is the first time you checkout or pull a new version of this repository to try the examples, you must call `./gradlew installRealmJava` from the top-level directory first. Otherwise the examples will not compile as they depend on all Realm artifacts being installed in `mavenLocal()`.
 
 Standalone examples can be [downloaded from website](https://realm.io/docs/java/latest/#getting-started).
 
+## Running Tests on a Device
+
+To run these tests you must have a device connected the the build computer and the `adb` command must be in your `PATH`
+
+1. Connect an Android device and verify that that the command `adb shell` starts a shell session on the device.
```
Comment (nhachicha) [inline README.md]: that that

Comment (nhachicha) [inline README.md]: why do we need to start a shell session?
a successful `adb devices` where the user can see his device will be enough IMO

Comment (bmeike) [inline README.md]: yep.  If you would prefer `adb devices` I will use that.

**Comment by beeender** [comment]
Merge this to `releases` branch instead?

See https://github.com/realm/realm-java/blob/master/CONTRIBUTING.md#branch-strategy

`edit` button on the top of this PR page -> change the base branch to `releases`

File: README.md [hunk 4]
```diff
@@ -70,38 +70,41 @@ In case you don't want to use the precompiled version, you can build Realm yours
...
- * If you are using OS X, you'd be better to add following lines to `~/.profile` (or `~/.zprofile` if the login shell is `zsh`) in order for Android Studio to see those environment variables.
+ * If you will be launching Android Studio from the OS X Finder, you should also run the following two commands:
 
     ```
     launchctl setenv ANDROID_HOME "$ANDROID_HOME"
     launchctl setenv ANDROID_NDK_HOME "$ANDROID_NDK_HOME"
     ```
 
- * And if you'd like to specify the location to store the archives of Realm's core, set `REALM_CORE_DOWNLOAD_DIR` environment variable. It enables you to keep core's archive when executing `git clean -xfd`.
+ * If you'd like to specify the location in which to store the archives of Realm's core, define the `REALM_CORE_DOWNLOAD_DIR` environment variable. It enables you to keep core's archive when executing `git clean -xfd`.
```
Comment (kneth) [inline README.md]: "Realm's core" -> "Realm Core" (after open sourcing core, it's a "product")

File: README.md [hunk 5]
```diff
@@ -131,16 +134,18 @@ That command will generate:
...
 ### Other Commands
 
  * `./gradlew tasks` will show all the available tasks
  * `./gradlew javadoc` will generate the Javadocs
  * `./gradlew monkeyExamples` will run the monkey tests on all the examples
  * `./gradlew installRealmJava` will install the Realm library and plugin to mavenLocal()
- * `./gradlew clean -PdontCleanJniFiles` will remove all generated files except for JNI related files. This saves recompilation time a lot.
- * `./gradlew connectedUnitTests -PbuildTargetABIs=$(adb shell getprop ro.product.cpu.abi)` will build JNI files only for the ABI which corresponds to the connected device.
+ * `./gradlew clean -PdontCleanJniFiles` will remove all generated files except for JNI related files. This reduces recompilation time a lot.
+ * `./gradlew connectedUnitTests -PbuildTargetABIs=$(adb shell getprop ro.product.cpu.abi)` will build JNI files only for the ABI which corresponds to the connected device.  These tests require a running Object Server (see below)
```
Comment (kneth) [inline README.md]: AFAIK, `connectedUnitTests` doesn't require a running Object Server. The `realm:testObjectServerDebugUnitTest` does require it.

Comment (beeender) [inline README.md]: `connectedUnitTests` does require the server unfortunately ... I failed to figure out how to make a separated task for integration tests.

Comment (cmelchior) [inline README.md]: I think can do that the same way I separated `connectedUnitTests` and `connectedBenchmarks`. It requires that the tests are placed in another package name though, otherwise we cannot configure the testrunner to ignore them. That is outside the scope of this PR though.

Comment (bmeike) [inline README.md]: empirically:

`./gradlew connectedBaseDebugAndroidTest`

runs without an Object Server.  However:

`./gradlew connectedObjectServerDebugAndroidTest`

does not.
